### DISCUSSION TITLE
Login & Sign up using Username & Password

### DISCUSSION THREAD

**Discussion by iamdineshkumar** [appwrite/appwrite#2324] [Q&A]
Hi
In many projects customers prefer username & password based authentication since they not allocate official email address to users as well as they require user management to create users and assign to teams to handle rights. Whether its possible to do?

**Comment by juliancuni** [comment]
About username login/signup, there is not much you can do unless you fork the project and make the appropriate changes (really cumbersome workaround).  As for user management creating and assigning users/teams/roles etc the only way you can work around is could functions. Or you can build your own backend with appwrite admin rights and implement there your custom authentication methods. But this undermines the BaaS purpose we all are here.  It would be nice if the team behind appwrite would take an approach to this requirement.
